### PR TITLE
Rdbc 349 compact database operation

### PR DESCRIPTION
**Pull request by reebhub**
RDBC-349 - CompactDatabaseOperation

Documentation of -
* The issue: an error while trying to compact a database that resides on a node other than the first online node.
* A temporary solution: explicitly identifying the database we want to compact.

### CODE REVIEW

**Comment by maximburyak** [comment]
@reebhub can you please make this changes only on the version of the 4.0 docs?
I'm looking at it now and I think that we added nothing new in 4.2 related to this issue

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+* CompactDatabaseOperation automatically runs **on the store's database**.  
+  If we try to compact **a different database**, the process will succeed only if the database 
+  resides **on the cluster's first online node**.  
+  Trying to compact a non-default database on a different node will fail with an error such as -  
+  **_"500 Internal Server Error : 
+  System.InvalidOperationException , 
+  Cannot compact database 'name' on node A, because it doesn't reside on this node."_**  
+  
+* To solve this, we can explicitly identify the database we want to compact by 
+  providing its name to a private CompactDatabaseOperation method.
```
Comment (ppekrol) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown]: `private CompactDatabaseOperation method` what do you mean by that?

File: Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Compact.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+                var compactOperation = new CompactDatabaseOperation(new Raven.Client.ServerWide.CompactSettings
+                {
+                    DatabaseName = DBToCompact,
+                    Documents = true,
+                    Indexes = indexNames
+                });
+
+                // Get request executor for our DB
+                var reqEx = store.GetRequestExecutor(DBToCompact);
```
Comment (ppekrol) [inline Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Compact.cs]: @ayende this is a bit inconvenient. I'm wondering if we should not expose `store.Maintenance.Server.ForNode("C")` ?

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown]: My awkward way of saying we can't send CompactDatabaseOperation the settings as in the previous examples or use some public method.
I can either be more accurate or simply give the example, how about:

* To solve this, we can explicitly identify the database we want to compact by providing its name to CompactDatabaseOperation as in the following example.

Comment (ppekrol) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown]: Sounds better, thanks

Comment (ayende) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown]: Yes, that is better

File: Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Compact.cs (hunk 2 above)
Comment (ayende) [reply Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Compact.cs]: Yes, I think so.

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/server-wide/compact-database.markdown]: done & pushed

**Pull request merged**